packages/annotations/src/format/annotation.js

```diff
@@ -64,19 +60,20 @@ export const annotation = {
 		return null;
 	},
 	__experimentalGetPropsForEditableTreePreparation( select, { richTextIdentifier, blockClientId } ) {
-		return {
-			annotations: select( 'core/annotations' ).__experimentalGetAnnotationsForRichText( blockClientId, richTextIdentifier ),
-		};
+		return select( 'core/annotations' ).__experimentalGetAnnotationsForRichText( blockClientId, richTextIdentifier );
 	},
-	__experimentalCreatePrepareEditableTree( props ) {
-		return ( formats, text ) => {
-			if ( props.annotations.length === 0 ) {
-				return formats;
+	__experimentalCreateFormatToValue( annotations ) {
+		return ( value ) => {
+			if ( annotations.length === 0 ) {
+				return value;
 			}
 
-			let record = { formats, text };
-			record = applyAnnotations( record, props.annotations );
-			return record.formats;
+			value = applyAnnotations( value, annotations );
+
+			return value;
```

Comment:
Why do you need the ability to replace text?

Reply:
I don't think I do, it's was just easier in rich-text to pass the whole value. I can easily change it to only allow changing the formats like `prepareEditableTree`.

Reply:
Maybe I should have commented this. We don't want to allow `prepareEditableTree` to change anything other than the formats. See https://github.com/WordPress/gutenberg/pull/11595#discussion_r231578231